src/uct/ib/base/ib_iface.c

```diff
@@ -406,60 +407,79 @@ ucs_status_t uct_ib_iface_get_device_address(uct_iface_h tl_iface,
     return UCS_OK;
 }
 
-static int uct_ib_iface_is_same_roce_version(uct_ib_roce_version_desc_t *local_roce_ver,
-                                             const uct_ib_address_t *remote_ib_addr)
+static inline uct_ib_device_roce_version_t
+uct_ib_address_flags_get_roce_version(uint8_t flags)
 {
-    int ret, remote_roce_ver;
+    ucs_assert(flags & UCT_IB_ADDRESS_FLAG_LINK_LAYER_ETH);
+
+    switch (flags >> ucs_ilog2(UCT_IB_ADDRESS_FLAG_LAST)) {
+    case UCT_IB_DEVICE_ROCE_V1:
+        return UCT_IB_DEVICE_ROCE_V1;
```

Comment:
after this refactoring i see we don't have to use switch case, just return `flags >> ucs_ilog2(UCT_IB_ADDRESS_FLAG_LAST)`